presto-main/src/main/java/com/facebook/presto/execution/scheduler/SourcePartitionedScheduler.java

```diff
@@ -74,6 +76,7 @@ public SourcePartitionedScheduler(
         this.splitBatchSize = splitBatchSize;
 
         this.partitionedNode = partitionedNode;
+        this.stageHasDefaultOutput = hasDefaultOutput(stage.getFragment().getRoot());
```

Comment:
add `defaultOutput` flag to `PlanFragment`. This will separate concerns (scheduler won't do planning/optimization job)